tools/fastp/fastp.xml

```diff
@@ -0,0 +1,332 @@
+                <param format="fastq" name="in2" type="data" optional="true" label="Input 2" argument="-I" help="Input FastQ file #2."/>
+                <section name="adapter_trimming_options" title="Adapter trimming options" expanded="False">
+                    <param name="disable_adapter_trimming" type="boolean" truevalue="-A" falsevalue="" checked="false" label="Disable adapter trimming" argument="-A" help="Adapter trimming is enabled by default. If this option is specified, adapter trimming is disabled."/>
+                    <param name="adapter_sequence1" type="text" optional="true" label="Adapter sequence for input 1" argument="--adapter_sequence" help="The adapter for read1. For SE data, if not specified, the adapter will be auto-detected. For PE data, this is used if R1/R2 are found not overlapped.">
+                        <sanitizer>
+                            <valid>
+                                <add value="A"/>
+                                <add value="T"/>
+                                <add value="C"/>
+                                <add value="G"/>
+                            </valid>
+                         </sanitizer>
+                    </param>
+                    <param name="adapter_sequence2" type="text" optional="true" label="Adapter sequence for input 2" argument="--adapter_sequence_r2" help="The adapter for read2 (PE data only). This is used if R1/R2 are found not overlapped. If not specified, it will be the same as adapter sequence value for input 1.">
+                        <sanitizer>
+                            <valid>
+                                <add value="A"/>
+                                <add value="T"/>
+                                <add value="C"/>
+                                <add value="G"/>
+                            </valid>
+                         </sanitizer>
+                    </param>
+                </section>
+                <section name="global_trimming_options" title="Global trimming options" expanded="False">
+                    <param name="trim_front1" type="integer" optional="true" label="Trim front for input 1" argument="-f" help="Trimming how many bases in front for read1, default is 0."/>
+                    <param name="trim_tail1" type="integer" optional="true" label="Trim tail for input 1" argument="-t" help="Trimming how many bases in tail for read1, default is 0."/>
+                    <param name="trim_front2" type="integer" optional="true" label="Trim front for input 2" argument="-F" help="Trimming how many bases in front for read2. If it's not specified, it will follow read1's settings."/>
+                    <param name="trim_tail2" type="integer" optional="true" label="Trim tail for input 2" argument="-T" help="Trimming how many bases in tail for read2. If it's not specified, it will follow read1's settings."/>
+                </section>
+            </when>
+        </conditional>
```

Comment:
If the sections below are not likely to be used by the user, I would hide them in an 'advanced_parameters' super section.